sys/fuchsia/fidlgen/main.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+var zirconLibs = []string{
+	"fuchsia-process",
+	"fuchsia-io",
+}
+
+func main() {
+	targetArch := os.ExpandEnv("${TARGETARCH}")
+	arch := ""
+	switch targetArch {
+	case "amd64":
+		arch = "x64"
+	case "arm64":
+		arch = "arm64"
+	default:
+		failf("unknown TARGETARCH %s", targetArch)
+	}
+
+	sourceDir := os.ExpandEnv("${SOURCEDIR}")
+	_, err := os.Stat(sourceDir)
+	if err != nil {
+		failf("cannot find SOURCEDIR %s", sourceDir)
+	}
+
+	for _, lib := range zirconLibs {
+		jsonPath := filepath.Join(
+			sourceDir,
+			"out",
+			arch,
+			"fidling/gen/zircon/public/fidl",
+			lib,
+			fmt.Sprintf("%s.fidl.json", lib),
+		)
+
+		if _, err := os.Stat(jsonPath); err != nil {
+			failf("cannot find %s", jsonPath)
+		}
+
+		txtPath := strings.Replace(lib, "fuchsia-", "fidl_", 1)
+		cmd := exec.Command("fidlgen",
+			"-generators", "syzkaller",
+			"-json", jsonPath,
+			"-output-base", txtPath,
+			"-include-base", txtPath,
+		)
+		fmt.Printf("%s\n", strings.Join(cmd.Args, " "))
+		err := cmd.Run()
+		if err != nil {
+			failf("fidlgen executed with error")
```

Comment:
Need to print the error too, and output.
For me this fails with "fidlgen executed with error", and what now?
Use `_, err := osutil.RunCmd(time.Minute, "", "fidlgen", ...`. The returned err already contains process output too, so you just print it.

Now I see that the problem is: `exec: "fidlgen": executable file not found in $PATH`. So that's something to fix too. Where should it come from?

Reply:
Sorry, a full fidlgen path is used now. Also the error/output is printed now (`err` returned by `osutil.RunCmd`).